pkg/asset/machines/aws/machines.go

```diff
@@ -153,61 +168,74 @@ func Machines(clusterID string, region string, subnets map[string]string, pool *
 	return machines, controlPlaneMachineSet, nil
 }
 
-func provider(clusterID string, region string, subnet string, instanceType string, root *aws.EC2RootVolume, imds aws.EC2Metadata, osImage string, zone, role, userDataSecret string, userTags map[string]string) (*machineapi.AWSMachineProviderConfig, error) {
-	tags, err := tagsFromUserTags(clusterID, userTags)
+func provider(in *machineProviderInput) (*machineapi.AWSMachineProviderConfig, error) {
+	tags, err := tagsFromUserTags(in.clusterID, in.userTags)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create machineapi.TagSpecifications from UserTags")
 	}
+	iamProfileName := fmt.Sprintf("%s-%s-profile", in.clusterID, in.role)
+	ebsRootSize := int64(in.root.Size)
+	ebsRootIOPS := int64(in.root.IOPS)
+	ebsRootEncrypted := true
 	config := &machineapi.AWSMachineProviderConfig{
 		TypeMeta: metav1.TypeMeta{
 			APIVersion: "machine.openshift.io/v1beta1",
 			Kind:       "AWSMachineProviderConfig",
 		},
-		InstanceType: instanceType,
+		InstanceType: in.instanceType,
 		BlockDevices: []machineapi.BlockDeviceMappingSpec{
 			{
 				EBS: &machineapi.EBSBlockDeviceSpec{
-					VolumeType: pointer.StringPtr(root.Type),
-					VolumeSize: pointer.Int64Ptr(int64(root.Size)),
-					Iops:       pointer.Int64Ptr(int64(root.IOPS)),
-					Encrypted:  pointer.BoolPtr(true),
-					KMSKey:     machineapi.AWSResourceReference{ARN: pointer.StringPtr(root.KMSKeyARN)},
```

Comment:
> Is this actually changing anything?

@JoelSpeed No, just providing the input structure to avoid the long amount of arguments.

> Why not use the pointer utils?

I would, but the linter job was reporting [as deprecated](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_installer/6371/pull-ci-openshift-installer-master-golint/1623845916333051904#1:build-log.txt%3A96-115):

```
A1019: pointer.Int64Ptr is deprecated: Use Int64 instead. (staticcheck)
					Iops:       pointer.Int64Ptr(int64(in.root.IOPS)),
					            ^
pkg/asset/machines/aws/machines.go:190:55: SA1019: pointer.StringPtr is deprecated: Use String instead. (staticcheck)
					KMSKey:     machineapi.AWSResourceReference{ARN: pointer.StringPtr(in.root.KMSKeyARN)},
(...)
```

Thoughts?

Reply:
FYI https://github.com/kubernetes/utils/pull/249

Reply:
I think they just renamed a bunch, so it's `pointer.String` no instead of `pointer.StringPtr`, they basically dropped `Ptr` from all the different function names

Reply:
@JoelSpeed you're completely right. That change passed to my eye. I just fixed that on [this review](https://github.com/openshift/installer/compare/cff35ba7dfe2b78984f49b73bfc20d11419fbe3c..c4d652e3d83bb4821eff4ed9b0a834c69c22ff1e). Thanks